palabos/source/plb_pressure_diff.cpp

```diff
@@ -0,0 +1,125 @@
+#include "palabos3D.h"
+#include "palabos3D.hh"
+#include <vector>
+#include <cmath>
+
+using namespace plb;
+using namespace std;
+
+typedef double T;
+#define DESCRIPTOR descriptors::D3Q19Descriptor
+
+// This function object returns a zero velocity, and a pressure which decreases
+//   linearly in x-direction. It is used to initialize the particle populations.
+class PressureGradient {
+public:
+        PressureGradient(T deltaP_, plint nx_) : deltaP(deltaP_), nx(nx_)
+        { }
+        void operator() (plint iX, plint iY, plint iZ, T& density, Array<T,3>& velocity) const
+        {
+                velocity.resetToZero();
+                density = 1. - deltaP*DESCRIPTOR<T>::invCs2 / (T)(nx-1) * (T)iX;
+        }
+private:
+        T deltaP;
+        plint nx;
+};
+
+void setupPressureDifferenceProblem( MultiBlockLattice3D<T,DESCRIPTOR>& lattice,
+                       OnLatticeBoundaryCondition3D<T,DESCRIPTOR>* boundaryCondition,
+                       MultiScalarField3D<int>& geometry, T deltaP)
+{
+        const plint nx = lattice.getNx();
+        const plint ny = lattice.getNy();
+        const plint nz = lattice.getNz();
+
+        Box3D inlet (0,0,      1,ny-2, 1,nz-2);
+        boundaryCondition->addPressureBoundary0N(inlet, lattice);
+        setBoundaryDensity(lattice, inlet, (T) 1.);
+
+        Box3D outlet(nx-1,nx-1, 1,ny-2, 1,nz-2);
+        boundaryCondition->addPressureBoundary0P(outlet, lattice);
+        setBoundaryDensity(lattice, outlet, (T) 1. - deltaP*DESCRIPTOR<T>::invCs2);
+
+        // Where "geometry" evaluates to 1 (solid boundary), use bounce-back.
+        defineDynamics(lattice, geometry, new BounceBack<T,DESCRIPTOR>(), 1);
+        // Where "geometry" evaluates to 2 (solid), use no-dynamics (which does nothing).
+        defineDynamics(lattice, geometry, new NoDynamics<T,DESCRIPTOR>(), 2);
+
+        initializeAtEquilibrium( lattice, lattice.getBoundingBox(), PressureGradient(deltaP, nx) );
+
+        lattice.initialize();
+        delete boundaryCondition;
+}
+
+void writeVTK_velocity(MultiBlockLattice3D<T,DESCRIPTOR>& lattice, string fname)
+{
+        VtkImageOutput3D<T> vtkOut(fname.c_str(), 1.);
+        vtkOut.writeData<3,float>(*computeVelocity(lattice), "velocity", 1.);
+}
+
+void writeVTK_density(MultiBlockLattice3D<T,DESCRIPTOR>& lattice, string fname)
+{
+        VtkImageOutput3D<T> vtkOut(fname.c_str(), 1.);
+        vtkOut.writeData<float>(*computeDensity(lattice), "density", 1.);
+}
+
+int main(int argc, char **argv)
+{
+        plbInit(&argc, &argv);
+        global::directories().setOutputDir("./");
+
+        if (argc < 2) {
+            pcout << "SimPhoNy-Palabos file-IO wrapper" << endl;
```

Comment:
Good catch. I'll write a more complete "program header" here. User is not supposed to directly run these executables, but still some kind of documentation of the command line parameters should be given.